mirdata/giantsteps_tempo.py

```diff
@@ -0,0 +1,331 @@
+                    > giantsteps-tempo-dataset-0b7d47ba8cae59d3535a02e3db69e2cf6d0af5bb/
+                    > audio/
+            and copy the folder to {}
+        """.format(
+        data_home
+    )
+
+    download_utils.downloader(
+        data_home,
+        remotes=REMOTES,
+        info_message=download_message,
+        force_overwrite=force_overwrite,
+        cleanup=cleanup,
+    )
+
+
+def validate(data_home=None, silence=False):
+    """Validate if a local version of this dataset is consistent
+    Args:
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+    Returns:
+        missing_files (list): List of file paths that are in the dataset index
+            but missing locally
+        invalid_checksums (list): List of file paths where the expected file exists locally
+            but has a different checksum than the reference
+    """
+    if data_home is None:
+        data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+    missing_files, invalid_checksums = utils.validator(
+        DATA.index, data_home, silence=silence
+    )
+    return missing_files, invalid_checksums
+
+
+def track_ids():
+    """Get the list of track IDs for this dataset
+    Returns:
+        (list): A list of track ids
+    """
+    return list(DATA.index.keys())
+
+
+def load(data_home=None):
+    """Load giantsteps_tempo dataset
+    Args:
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+    Returns:
+        (dict): {`track_id`: track data}
+    """
+    if data_home is None:
+        data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+    giantsteps_key_data = {}
+    for tempo in track_ids():
+        giantsteps_key_data[tempo] = Track(tempo, data_home=data_home)
+    return giantsteps_key_data
+
+
+def load_genre(path):
+    """Load genre data from a file
+    Args:
+        path (str): path to metadata annotation file
+    Returns:
+        (str): loaded genre data
+    """
+    if path is None:
+        return None
+
+    with open(path) as json_file:
+        annotation = jams.load(json_file)
+
+    return annotation.search(namespace='tag_open')[0]['data'][0].value
+
+
+def load_tempo(tempo_path):
+    """Load giantsteps_tempo tempo data from a file ordered by confidence
+    Args:
+        tempo_path (str): path to tempo annotation file
+    Returns:
+        (list of utils.TempoData): loaded tempo data
+    """
+    if tempo_path is None:
+        return None
+
+    if not os.path.exists(tempo_path):
+        raise IOError("tempo_path {} does not exist".format(tempo_path))
+
+    with open(tempo_path) as json_file:
+        annotation = jams.load(json_file)
+
+    tempos = annotation.search(namespace='tempo')[0]['data']
+    return [
+        utils.TempoData(tempo.duration, tempo.confidence, tempo.value, tempo.time)
```

Comment:
The way we've been using data types is the following:

instead of returning a list of let's say `BeatData`, we return the `BeatData` tuple containing arrays with the expected beat times and beat positions, like this:

`return beat_data = utils.BeatData(np.array(beat_times), np.array(beat_positions))`.

In your case, I think what you want to do is:

`return utils.TempoData(np.array(tempo.time), np.array(tempo.duration), np.array(tempo.value), np.array( tempo.confidence))` 

or similar

Reply:
done